### ISSUE TITLE
Script syntax errors in Chrome do not include filename

### ISSUE THREAD

**Issue by greglittlefield-wf** [newrelic/newrelic-browser-agent#836]
[NOTE]: # ( ^^ Provide a general summary of the issue in the title above. ^^ )

## Description
For script errors resulting from syntax errors, only the error message seem to be reported, and not any information about the filename, at least in Chrome. I have not confirmed whether this issue is Chrome specific.

For example, one of these errors might show up in the New Relic browser errors view as
> Javascript error: Error occurred with the message "Unexpected end of input"

With only the following fields providing information about the error (stack trace is empty, no other fields containing the file name):
- errorClass: SyntaxError
- errorMessage: Unexpected end of input

This may be the same issue described here, which mentions this issue being Chrome specific: https://forum.newrelic.com/s/hubtopic/aAX8W0000008YQXWA2/script-error-information

## Steps to Reproduce
Here's an example of what errors look like when passed to window error handlers:
```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
  </head>
  <body>
    <script>
      window.addEventListener("error", function (event) {
        console.log('window error handler', {
          event: event,
          "event.error": event.error,
          "event.error instanceof Error": event.error instanceof Error,
        });
      });
    </script>

    <!-- Fail with syntax error -->
    <script>
      -
    </script>

    <!-- Normal uncaught error -->
    <script>
      throw new Error("Test uncaught error");
    </script>
  </body>
</html>
```

It looks like information is extracted from window error events in this castErrorEvent call here: https://github.com/newrelic/newrelic-browser-agent/blob/3e3b8102aec2105593b5fec07445400107b9941f/src/features/jserrors/instrument/index.js#L60, which only extracts filename information from error events when the provided error is not `instanceof Error`: https://github.com/newrelic/newrelic-browser-agent/blob/3e3b8102aec2105593b5fec07445400107b9941f/src/features/jserrors/instrument/index.js#L130-L140

In Chrome, it seems that the `event.error` does not seem to have any information on the filename (including in the stack) for syntax errors, unlike for other uncaught errors. However, it is available on the `event.filename`.

> ![chrome-console-output](https://github.com/newrelic/newrelic-browser-agent/assets/1713967/74bbc2d8-a4fe-4ae3-b79d-fa439a065581)

So, it makes sense to me that it wouldn't get reported.

In Firefox, on the other hand, more information is available on `event.error`:

> ![firefox-console-output](https://github.com/newrelic/newrelic-browser-agent/assets/1713967/77216d5f-cb26-451a-af40-68c726c94a70)

## Expected Behavior

I would expect at least the file name to be included in the reported data.

#### Browsers where data has been reported without filenames
Chrome 119, Chrome 120

#### Steps to reproduce
OS: macOS 13.6.2
Chrome: 120.0.6099.71, 119.0.6045.199
Firefox: 119.0.1

**Comment by metal-messiah** [comment]
We've taken this request under advisement and added the work to our backlog for future handling.  Thanks for submitting this and pointing out this behavior!

**Comment by patrickhousley** [comment]
The feature for this was released in 1.252.0. Please refer to the [release notes](https://docs.newrelic.com/docs/release-notes/new-relic-browser-release-notes/browser-agent-release-notes/browser-agent-v1.252.0/) for more information. If you are still having issues after upgrading your agent, please let us know.
